### PR TITLE
Idempotency

### PR DESCRIPTION
**Pull request by tdfischer**

### CODE REVIEW

**Comment by tdfischer** [comment]
+r

File: test/amortizer.js [hunk 1]
```diff
@@ -7,67 +15,68 @@ var assert     = require('assert');
...
+    return db.knex.migrate.latest(db.conf).then(function() {
       startTime = Date.now();
-      amortizer._instances[token.get('token')] = {
-        lastCheckedBalance: 10,
+      startBalance = 10;
+      amortizer._instances[token] = {
+        lastCheckedBalance: startBalance,
         lastChargedTime: startTime
       }
-      token.getBalance().then(function(balance) {
```
Comment (wilsonianb) [inline test/amortizer.js]: Is this code (lines 25-29) necessary?

**Comment by stevenzeiler** [comment]
I'll take idempotent potables for 300

File: test/amortizer.js [hunk 2]
```diff
@@ -7,67 +15,68 @@ var assert     = require('assert');
...
+    return token.destroy().then(function() {
+      return db.knex.migrate.rollback(db.conf);
     });
   });
 
-  it('should check a token\'s balance from the database', function(done) {
-    new Token({ token: uuid.v4() }).save().then(function(token_) {
-      amortizer.checkTokenBalance(token_).then(function(balance) {
+  it('should check a token\'s balance from the database', function() {
+    return new Token({ token: uuid.v4() }).save().then(function(token_) {
```
Comment (wilsonianb) [inline test/amortizer.js]: Why not use `token` from `beforeEach`?

Comment (wilsonianb) [inline test/amortizer.js]: Ah, because it's balance is amortizing and is hard to test. nvm

File: test/amortizer.js [hunk 3]
```diff
@@ -7,67 +15,68 @@ var assert     = require('assert');
...
-
-    amortizer.chargeToken(token).then(function(balance) {
-      assert(startBalance - Math.ceil((Date.now() - startTime) / 100) <= balance);
-      startBalance = balance;
-      done();
+  it('should charge a running instance\'s balance', function() {
+    return Promise.delay(10).then(function() {
+      return amortizer.chargeToken(token).then(function(balance) {
+        assert(startBalance - Math.ceil((Date.now() - startTime) / 100) <= balance);
+        startBalance = balance;
```
Comment (wilsonianb) [inline test/amortizer.js]: This line can go.

File: test/amortizer.js (hunk 1 above)
Comment (tdfischer) [inline test/amortizer.js]: No idea. I wanted to modify the tests as little as needed so I kept it in there.

File: test/amortizer.js (hunk 3 above)
Comment (tdfischer) [inline test/amortizer.js]: Fix pushed.

**Comment by wilsonianb** [comment]
LGTM

**Comment by codius-montecarlo** [comment]
The build is good and I'm seeing 1 +1s from the following folks:

* wilsonianb

As such, I'll attempt to merge and close this pull request.

**Pull request merged**